nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/sensitive/ProtectedNiFiProperties.java

```diff
@@ -53,17 +57,17 @@
     public static final List<String> DEFAULT_SENSITIVE_PROPERTIES = new ArrayList<>(asList(SECURITY_KEY_PASSWD,
             SECURITY_KEYSTORE_PASSWD, SECURITY_TRUSTSTORE_PASSWD, SENSITIVE_PROPS_KEY, PROVENANCE_REPO_ENCRYPTION_KEY));
 
-    public ProtectedNiFiProperties() {
-        this(new StandardNiFiProperties());
-    }
+    // Default encryption key, hex encoded.
+    private String defaultKeyHex;
```

Comment:
We should change this to the default provider rather than loading the raw key and maintaining it throughout the lifecycle of this object.

Reply:
That's a good idea.  I've replaced the string member with an instance of SensitivePropertyProvider.  

I've also added constructors to that accept strings and create SSPP instances as a convenience for the caller.